app/models/manageiq/providers/embedded_ansible/automation_manager/credential.rb

```diff
@@ -26,6 +26,18 @@ def self.raw_create_in_provider(manager, params)
     create!(create_params)
   end
 
+  def self.password_attribute_keys
+    self::API_ATTRIBUTES.map do |k, v|
+      k if v[:type] == :password
+    end.compact
+  end
+
+  def self.encrypt_queue_params(params)
+    encrypted_params = params.slice(*password_attribute_keys)
+    encrypted_params.transform_values! { |v| ManageIQ::Password.try_encrypt(v) }
```

Comment:
Just putting this as a note to anyone else that sees this and is confused:

`ManageIQ::Password.try_encrypt` will only encrypt if the field if it hasn't been encrypted already (the check is done via `ManageIQ::Password.split`), and will just use the existing value if it has already been encrypted.  This is more important not for here, but in `PasswordMixin` since there isn't any `decrypt` step in this code, because it isn't required to save/update.